src/v/cluster/data_policy_frontend.cc

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "cluster/data_policy_frontend.h"
+
+#include "cluster/cluster_utils.h"
+
+#include <optional>
+
+namespace cluster {
+
+data_policy_frontend::data_policy_frontend(
+  ss::sharded<controller_stm>& stm, ss::sharded<ss::abort_source>& as) noexcept
+  : _stm(stm)
+  , _as(as) {}
+
+ss::future<std::error_code> data_policy_frontend::set_data_policy(
+  model::topic_namespace topic,
+  v8_engine::data_policy dp,
+  model::timeout_clock::time_point tout) {
+    update_data_policy_cmd_data cmd_data{.dp = std::move(dp)};
+    update_data_policy_cmd cmd(std::move(topic), std::move(cmd_data));
+    return replicate_and_wait(_stm, _as, std::move(cmd), tout);
+}
+
+ss::future<std::error_code> data_policy_frontend::clear_data_policy(
```

Comment:
I don't think this one needs to match the command name since it is a higher level interface: its removing all data policies, not one by identity.

Reply:
I am confused a little bit
Why for update names must be sync, but for delete - not